app/src/main/java/fr/free/nrw/commons/auth/LoginActivity.java

```diff
@@ -133,14 +136,35 @@ public void onCreate(Bundle savedInstanceState) {
         usernameEdit.addTextChangedListener(textWatcher);
         passwordEdit.addTextChangedListener(textWatcher);
         twoFactorEdit.addTextChangedListener(textWatcher);
-        
+
         if (ConfigUtils.isBetaFlavour()) {
             loginCredentials.setText(getString(R.string.login_credential));
         } else {
             loginCredentials.setVisibility(View.GONE);
         }
     }
 
+    // Return true is system wide dark theme is enabled else false
+    private boolean getSystemDefaultThemeBool(String theme) {
+        switch (theme) {
+            case "Dark":
+                return true;
+            case "Default":
+                return getSystemDefaultThemeBool(getSystemDefaultTheme());
+            default:
+                return false;
+        }
+    }
+
+    // Returns the default system wide theme
+    private String getSystemDefaultTheme() {
+        if ((getResources().getConfiguration().uiMode &
```

Comment:
If you just returned true or false here you could simplify this method down to
```
return (getResources().getConfiguration().uiMode & Configuration.UI_MODE_NIGHT_MASK) == Configuration.UI_MODE_NIGHT_YES;

Reply:
Ah but it is used as the default value for the store. Well a method could be extracted certainly, "isDeviceInNightMode"

Reply:
Done!